### PR TITLE
[iOS] Update logic to create new window

### PR DESCRIPTION
**Pull request by binhndicts**

### CODE REVIEW

File: src/WebViewTypes.ts [hunk 1]
```diff
@@ -245,6 +245,7 @@ export interface CommonNativeWebViewProps extends ViewProps {
   onMessage: (event: WebViewMessageEvent) => void;
   onShouldStartLoadWithRequest: (event: WebViewNavigationEvent) => void;
   onShouldCreateNewWindow: (event: WebViewNavigationEvent) => void;
+  onWebviewClosed?: (event: WebViewNavigationEvent) => void;
```
Comment (kazkondo) [inline src/WebViewTypes.ts:248]: it should be "onWebViewClosed"

File: ios/RNCWebView.m [hunk 2]
```diff
@@ -89,6 +90,13 @@ @implementation RNCWebView
   BOOL decelerating;
   BOOL dragging;
   BOOL scrollingToTop;
+  BOOL initiated;
+}
+
+- (void)webViewDidClose:(WKWebView *)webView {
+  if (_onWebviewClosed) {
```
Comment (kazkondo) [inline ios/RNCWebView.m:97]: It should be "_onWebViewClosed"

File: ios/RNCWebViewManager.m [hunk 3]
```diff
@@ -333,35 +334,32 @@ - (RNCWebView*)webView:(__unused RNCWebView *)webView
...
-  // } else {
-  //   RCTLogWarn(@"Did not receive response to shouldCreateNewWindow in time, defaulting to YES");
-  //   newWindow = [RNCWebView new];
-  //   [newWindow setupConfiguration:configuration];
-  //   return newWindow;
-  // }
-  return nil;
+  // Block the main thread for a maximum of 250ms until the JS thread returns
+  if ([createNewWindowCondition lockWhenCondition:0 beforeDate:[NSDate dateWithTimeIntervalSinceNow:.25]]) {
+    [createNewWindowCondition unlock];
```
Comment (kazkondo) [inline ios/RNCWebViewManager.m:339]: Does this fail if it is over 250ms?

Comment (binhndicts) [reply ios/RNCWebViewManager.m:339]: No. default value is YES

**Pull request merged**